src/Nethermind/Nethermind.Db/CompressingDb.cs

```diff
@@ -51,7 +53,37 @@ public void Set(ReadOnlySpan<byte> key, byte[]? value, WriteFlags flags = WriteF
 
                 public void PutSpan(ReadOnlySpan<byte> key, ReadOnlySpan<byte> value, WriteFlags flags = WriteFlags.None)
                 {
-                    _wrapped.PutSpan(key, Compress(value, stackalloc byte[value.Length]), flags);
+                    if (!value.EndsWith(EmptyCodeHashStorageRoot))
+                    {
+                        _wrapped.PutSpan(key, value, flags);
+                        return;
+                    }
+
+                    int storedLength = value.Length - EmptyCodeHashStorageRoot.Length;
+                    int compressedLength = storedLength + PreambleLength;
+
+                    if (compressedLength <= EOACompressingDb.StackAllocThreshold)
+                    {
+                        Span<byte> tmp = stackalloc byte[compressedLength];
```

Comment:
Stackalloc to fixed const `StackAllocThreshold` and slice to `compressedLength`; then add `[SkipLocalsInit]` to method

e.g.
```csharp
Span<byte> tmp = stackalloc byte[EOACompressingDb.StackAllocThreshold][..compressedLength];
```